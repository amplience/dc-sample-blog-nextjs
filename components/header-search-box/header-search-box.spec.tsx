/* eslint-env jest */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HeaderSearchBox from './header-search-box';

describe('HeaderSearchBox', () => {
  it('should render a header search box', async () => {
    const props = {
      placeholderText: 'Search box placeholder text'
    };

    const component = ShallowRenderer.createRenderer();
    component.render(<HeaderSearchBox {...props} />);
    expect(component.getRenderOutput()).toMatchSnapshot();
  });
});
