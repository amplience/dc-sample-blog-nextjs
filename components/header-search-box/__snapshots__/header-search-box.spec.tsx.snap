// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearchBox should render a header search box 1`] = `
<React.Fragment>
  <UNDEFINED
    submit={
      <svg
        className="jsx-2416404770"
        height="16"
        viewBox="0 0 18 18"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="jsx-2416404770"
          fill="none"
          fillRule="evenodd"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.67"
          transform="translate(1 1)"
        >
          <circle
            className="jsx-2416404770"
            cx="7.11"
            cy="7.11"
            r="7.11"
          />
          <path
            className="jsx-2416404770"
            d="M16 16l-3.87-3.87"
          />
        </g>
      </svg>
    }
    translations={
      Object {
        "placeholder": "Search box placeholder text",
      }
    }
  />
  <JSXStyle
    dynamic={
      Array [
        "#cccccc",
        "#cccccc",
        "#cccccc",
        "#cccccc",
        "#666666",
        "#cccccc",
        "#cccccc",
        "736px",
      ]
    }
    id="509995170"
  >
    input[type='search']::-ms-clear{display:none;width:0;height:0;}input[type='search']::-ms-reveal{display:none;width:0;height:0;}input[type='search']::-webkit-search-decoration,input[type='search']::-webkit-search-cancel-button,input[type='search']::-webkit-search-results-button,input[type='search']::-webkit-search-results-decoration{display:none;}.ais-SearchBox-form{position:relative;}.ais-SearchBox{height:57px;width:460px;max-width:100%;}.ais-SearchBox-input{height:57px;background-color:#4a515a;color:#cccccc;border-radius:3px;font-size:18px;width:100%;border:none;padding-left:56px;-webkit-appearance:none;-moz-appearance:none;-webkit-appearance:none;-moz-appearance:none;appearance:none;}.ais-SearchBox-input:focus,.ais-SearchBox-submit:focus,.ais-SearchBox-reset:focus{outline:none;}.ais-SearchBox .ais-SearchBox-input::-webkit-input-placeholder{color:#cccccc;font-style:italic;opacity:1;}.ais-SearchBox .ais-SearchBox-input::-moz-placeholder{color:#cccccc;font-style:italic;opacity:1;}.ais-SearchBox .ais-SearchBox-input:-ms-input-placeholder{color:#cccccc;font-style:italic;opacity:1;}.ais-SearchBox .ais-SearchBox-input::placeholder{color:#cccccc;font-style:italic;opacity:1;}.ais-SearchBox-input:-ms-input-placeholder{color:#cccccc;}.ais-SearchBox-input::-ms-input-placeholder{color:#cccccc;}.ais-SearchBox-input:focus{background-color:#fff;color:#666666;border:1px solid #039b35;}.ais-SearchBox-submit{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;position:absolute;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-transform:translateY(-50%);-ms-transform:translateY(-50%);transform:translateY(-50%);left:0;top:50%;height:100%;color:#cccccc;width:64px;}.ais-SearchBox-reset{right:0;color:#cccccc;}.ais-SearchBox-reset,.ais-SearchBox-loadingIndicator,.ais-SearchBox-submit{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-appearance:none;-moz-appearance:none;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:100%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:absolute;top:50%;-webkit-transform:translateY(-50%);-ms-transform:translateY(-50%);transform:translateY(-50%);width:48px;border:0;background:none;}@media (max-width:736px){.ais-SearchBox-input{padding-right:42px;text-overflow:ellipsis;}}
  </JSXStyle>
</React.Fragment>
`;
